txcache/globalAccountBreadcrumbsCompiler_test.go

```diff
@@ -0,0 +1,578 @@
+package txcache
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/stretchr/testify/require"
+)
+
+func requireEqualGlobalAccountsBreadcrumbs(
+	t *testing.T,
+	expected map[string]*globalAccountBreadcrumb,
+	actual map[string]*globalAccountBreadcrumb,
+) {
+	require.Equal(t, len(expected), len(actual))
+	for account, globalBreadcrumb := range expected {
+		_, ok := actual[account]
+		require.True(t, ok)
+		require.Equal(t, globalBreadcrumb, actual[account])
+	}
+}
+
+func Test_newGlobalAccountBreadcrumbsCompiler(t *testing.T) {
+	t.Parallel()
+
+	gabc := newGlobalAccountBreadcrumbsCompiler()
+	require.NotNil(t, gabc)
+}
+
+func Test_shouldWorkConcurrently(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		if r != nil {
+			require.Fail(t, "should not panic")
+		}
+	}()
+
+	gabc := newGlobalAccountBreadcrumbsCompiler()
+	numOfBlocks := 10
+
+	wg := sync.WaitGroup{}
+	wg.Add(2 * numOfBlocks)
+
+	for i := 1; i <= numOfBlocks; i++ {
+		tb := newTrackedBlock(uint64(i), []byte(fmt.Sprintf("hash%d", i-1)), []byte("rootHash0"), []byte(fmt.Sprintf("prevHash%d", i-1)))
+		go func(tb *trackedBlock) {
+			defer wg.Done()
+
+			gabc.updateGlobalBreadcrumbsOnAddedBlockOnProposed(tb)
+
+			time.Sleep(100 * time.Millisecond)
+		}(tb)
+
+		go func(tb *trackedBlock) {
+			defer wg.Done()
+
+			err := gabc.updateGlobalBreadcrumbsOnRemovedBlockOnExecuted(tb)
+			require.NoError(t, err)
+
+			time.Sleep(100 * time.Millisecond)
+		}(tb)
+	}
+
+	wg.Wait()
+}
+
+func Test_shouldWork(t *testing.T) {
```

Comment:
I think (not sure) that function name pollutes the test namespace of the package. Maybe find a more specific name?